src/utils.ts

```diff
@@ -13,7 +13,18 @@ export class ChainUtils {
     } else if (network.config.chainId === 100) {
       return Chain.XDAI;
     } else {
-      throw `Unsexpected network ${network.config.chainId}`;
+      throw `Unexpected network ${network.config.chainId}`;
```

Comment:
nit: (sorry missed this before)
```suggestion
      throw new Error(`Unexpected network ${network.config.chainId}`);
```

In general throwing errors is preferable to `string` as it provides properties like `message` (the specified error message) and more importantly `stacktrace`, which is a captured stacktrace from where the error is thrown and makes debugging easier.

Consider the code:
```js
function foo() { bar(); }
function bar() { baz(); }
function baz() { throw ... }
foo();
```

Throwing a `string`:
```
$ node ./foo_string.js

/var/home/nlordell/Developer/gp-v2-contracts/foo_string.js:3
function baz() { throw "error" }
                 ^
error
(Use `node --trace-uncaught ...` to show where the exception was thrown)
```

Throwing an `Error`:
```
$ node ./foo_error.js   
/var/home/nlordell/Developer/gp-v2-contracts/foo_error.js:3
function baz() { throw new Error("error") }
                 ^

Error: error
    at baz (/var/home/nlordell/Developer/gp-v2-contracts/foo_error.js:3:24)
    at bar (/var/home/nlordell/Developer/gp-v2-contracts/foo_error.js:2:18)
    at foo (/var/home/nlordell/Developer/gp-v2-contracts/foo_error.js:1:18)
    at Object.<anonymous> (/var/home/nlordell/Developer/gp-v2-contracts/foo_error.js:4:1)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:72:12)
    at internal/main/run_main_module.js:17:47
```

Reply:
If you don't want to show the stack trace, you can always catch errors at a top level and log only the message, and then print the stack trace to some debug logs (if using the `debug` package for example).

This also has the advantage that if there is some uncaught exception in some fiber that isn't `await`-ed you will get a stack trace.